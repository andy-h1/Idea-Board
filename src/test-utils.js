import React from 'react';
import { render } from '@testing-library/react';
import { IdeaContext } from './contexts/IdeaContext';

const AllTheProviders = ({ children, customContext }) => (
  <IdeaContext.Provider
    value={{
      ideas: [],
      sortIdea: () => null,
      ...customContext
    }}
  >
    {children}
  </IdeaContext.Provider>
);

const customRender = (ui, customContext, options) => {
  /* eslint-disable react/display-name */
  return render(ui, {
    wrapper: () => <AllTheProviders customContext={customContext}>{ui}</AllTheProviders>,
    ...options
  });
};

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };
