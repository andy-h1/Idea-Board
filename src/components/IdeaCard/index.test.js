import React from 'react';
import userEvent from '@testing-library/user-event';
import { render } from '../../test-utils';
import { IdeaCard } from '.';

const idea = {
  title: 'League of Legends',
  description: 'MOBA',
  id: '1',
  time: new Date().toLocaleTimeString()
};

describe('Idea Card Component', () => {
  it('should render', () => {
    const { container } = render(<IdeaCard />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it.only('should render form when user clicks edit', () => {
    const { getByTestId } = render(<IdeaCard />, { idea });
    userEvent.click(getByTestId('editButton'));
    expect(getByTestId('updateIdeaForm'));
  });
  // user clicks the 'edit' button
  // card should show two input fields
  // input fields should show current data

  it.todo('should render updated time when card has been updated');
  // user clicks 'Update Idea' button card
  // time property updated
  // on the card time created changed to time updated
  // displays the new time

  it.todo('should render updated changes onto the card');
  // user adds changes to either title or description inputs
  // user clicks 'Update Idea' button card
  // card rerenders with updated changes

  it.todo('should show the users characters remaining in textarea');
  // user types in description text area
  // currently 120+ in description area
  // card shows current characters remaining to user

  it.todo('should render data on the idea card');
});
