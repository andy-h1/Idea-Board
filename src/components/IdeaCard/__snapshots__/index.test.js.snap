// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Idea Card Component should render 1`] = `
<form
  class="sc-bdnylx hBwQcZ"
  data-testid="ideaCardCard"
>
  <h3
    class="sc-eCApGN jmaljw"
  >
    League of Legends
  </h3>
  <p
    class="sc-jSFkmK ijWLCw"
  >
    MOBA
  </p>
  <p
    class="sc-gKAblj kmRkkF"
  >
    Time 
    created
    : 
    00:00:00
  </p>
  <div
    class="sc-iCoHVE gTSoTY"
  >
    <button
      class="sc-fujyUd fxUbUe"
      data-testid="ideaCardRemoveButton"
      type="button"
    >
      Delete
    </button>
    <button
      class="sc-fujyUd fxUbUe"
      data-testid="ideaCardEditButton"
      type="button"
    >
      Edit
    </button>
  </div>
</form>
`;
